Makefile

```diff
@@ -113,7 +113,22 @@ vet:
 	curl -L -o $@ https://github.com/nspcc-dev/.github/raw/master/.golangci.yml
 
 lint: .golangci.yml
-	@golangci-lint run
+	-@golangci-lint run ./...
+	-@cd ./examples/engine/ && golangci-lint run
+	-@cd ./examples/events/ && golangci-lint run
+	-@cd ./examples/iterator/ && golangci-lint run
+	-@cd ./examples/nft-d/ && golangci-lint run
+	-@cd ./examples/nft-nd/ && golangci-lint run
+	-@cd ./examples/nft-nd-nns/ && golangci-lint run
+	-@cd ./examples/oracle/ && golangci-lint run
+	-@cd ./examples/runtime/ && golangci-lint run
+	-@cd ./examples/storage/ && golangci-lint run
+	-@cd ./examples/timer/ && golangci-lint run
+	-@cd ./examples/token/ && golangci-lint run
+	-@cd ./examples/zkp/cubic_circuit/ && golangci-lint run
+	-@cd ./examples/zkp/xor_compat/ && golangci-lint run
+	-@cd ./scripts/ && golangci-lint run ./...
+	-@cd ./pkg/interop/ && golangci-lint run ./...
```

Comment:
@roman-khimov, need your help with this target, I spent too much time on it, and it's my best attempt.

Reply:
```
diff --git a/Makefile b/Makefile
index 82194ce04..0dcd98393 100644
--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,9 @@ IMAGE_REPO=nspccdev/neo-go
 
 DISABLE_NEOTEST_COVER=1
 
+ROOT_DIR:=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+GOMODDIRS=$(dir $(shell find $(ROOT_DIR) -name go.mod))
+
 # All of the targets are phony here because we don't really use make dependency
 # tracking for files
 .PHONY: build $(BINARY) deps image docker/$(BINARY) image-latest image-push image-push-latest clean-cluster \
@@ -113,7 +116,9 @@ vet:
        curl -L -o $@ https://github.com/nspcc-dev/.github/raw/master/.golangci.yml
 
 lint: .golangci.yml
-       @golangci-lint run
+       @for dir in $(GOMODDIRS); do \
+               (cd "$$dir" && golangci-lint run --config $(ROOT_DIR)/$< | sed -r "s,^,$$dir," | sed -r "s,^$(ROOT_DIR),,") \
+       done
 
 fmt:
        @gofmt -l -w -s $$(find . -type f -name '*.go'| grep -v "/vendor/")
```
A bit lazy in seds, but should work.

Reply:
But the problem with this script is that it will fail on the first unlinted file. My original aim was to output all errors at ones.

Reply:
Why would it?

Reply:
It currently outputs
```
examples/timer/timer.go:16:51  godot   Comment should end in a period
examples/timer/timer.go:18:64  godot   Comment should end in a period
examples/timer/timer.go:20:48  godot   Comment should end in a period
examples/timer/timer.go:12:7   unused  const `defaultTicks` is unused
examples/timer/timer.go:28:6   unused  func `_deploy` is unused
examples/runtime/runtime.go:10:64  godot   Comment should end in a period
examples/runtime/runtime.go:25:6   unused  func `_deploy` is unused
examples/token/nep17/nep17.go:11:30   godot       Comment should end in a period
examples/token/nep17/nep17.go:38:62   godot       Comment should end in a period
examples/token/nep17/nep17.go:43:58   godot       Comment should end in a period
examples/token/nep17/nep17.go:99:22   whitespace  unnecessary leading newline
examples/token/nep17/nep17.go:121:22  gosimple    S1002: should omit comparison to bool constant, can be simplified to `minted.(bool)`
examples/storage/storage.go:8:50   godot  Comment should end in a period
examples/storage/storage.go:14:73  godot  Comment should end in a period
examples/iterator/iterator.go:10:6  unused  func `_deploy` is unused
examples/zkp/cubic_circuit/main.go:1  typecheck  : # github.com/nspcc-dev/neo-go/examples/zkp/cubic [github.com/nspcc-dev/neo-go/examples/zkp/cubic.test]
examples/zkp/cubic_circuit/./main_test.go:318:6: declared and not used: i
examples/zkp/xor_compat/verify.go:46:116  godot     Comment should end in a period
examples/zkp/xor_compat/verify.go:71:2    intrange  for loop can be changed to use an integer range (Go 1.22+)
examples/zkp/xor_compat/verify.go:75:2    intrange  for loop can be changed to use an integer range (Go 1.22+)
examples/nft-nd-nns/nns.go:63:2        unused       const `defaultRegisterPrice` is unused
examples/nft-nd-nns/nns.go:82:6        unused       func `_deploy` is unused
examples/nft-nd-nns/nns.go:494:9       gosimple     S1009: should omit nil check; len() for github.com/nspcc-dev/neo-go/pkg/interop.Hash160 is defined as zero
examples/nft-nd-nns/nns.go:558:2       intrange     for loop can be changed to use an integer range (Go 1.22+)
examples/nft-nd-nns/nns_test.go:425:9  staticcheck  SA4010: this result of append is never used, except maybe in other appends
examples/nft-nd/nft.go:29:65  godot  Comment should end in a period
examples/engine/engine.go:7:79   godot  Comment should end in a period
examples/engine/engine.go:13:79  godot  Comment should end in a period
examples/engine/engine.go:19:84  godot  Comment should end in a period
ERRO Running error: context loading failed: no go files to analyze: running `go mod tidy` may solve the problem 
pkg/interop/native/neo/neo.go:134:1                    whitespace  unnecessary trailing newline
pkg/interop/native/ledger/transaction_signer.go:91:35  godot       Comment should end in a period
```
for me.

Reply:
Oh, sorry.

Reply:
Added, but: your script finds issues that GA can't find, specifically, for nep17 token example:
```
examples/token/nep17/nep17.go:11:30   godot       Comment should end in a period
examples/token/nep17/nep17.go:38:62   godot       Comment should end in a period
examples/token/nep17/nep17.go:43:58   godot       Comment should end in a period
examples/token/nep17/nep17.go:99:22   whitespace  unnecessary leading newline
examples/token/nep17/nep17.go:121:22  gosimple    S1002: should omit comparison to bool constant, can be simplified to `minted.(bool)`
```

I suggested that nested directories linting isn't working, but I checked with NeoGo linter, and turns out it works.